test/Kestrel.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -169,6 +172,66 @@ public Task BadRequestIfHostHeaderDoesNotMatchRequestTarget(string requestTarget
             Assert.Equal(LogLevel.Information, maxLogLevel);
         }
 
+        [Fact]
+        public async Task HostHeaderAcceptedWithPeriod()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host: .",
+                        "Connection: close\r\n\r\n");
+
+                    await client.ReceiveStartsWith("HTTP/1.1 200");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task HttpRequestSplittingFails()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET /\x0D\0x0ALocation:http://www.contoso.com/ HTTP/1.1",
+                        "Host:\r\n\r\n");
+
+                    await client.ReceiveStartsWith("HTTP/1.1 400");
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("[0x00]", 400)]
```

Comment:
I'm adding `"[0x00]"` to `HttpParsingData.MethodWithNonTokenCharData`, then this will be tested via `BadHttpRequestTests.TestInvalidRequestLines` as well.